src/server/worker/datum/util.go

```diff
@@ -0,0 +1,59 @@
+package datum
+
+import (
+	"time"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/pachyderm/pachyderm/src/client/pps"
+)
+
+// MergeStats merges two stats.
+func MergeStats(x, y *Stats) error {
```

Comment:
I think this would be more straightforward if it returned a new Stats object.

Reply:
Looking more at this and the method below: if it is really that important to mutate a Stats object vs create a new one then this should be a method so it's obvious which stats object will be mutated.

Reply:
I agree with the suggested change, and you should be able to just add methods to the generated struct? Is this just bad practice, or is there some reason why this wouldn't work?

Reply:
There's no reason why it wouldn't work. Just do it in separate file (obviously not the generated one).